mantidproject/mantid
# Efixed mode for paalman pings corrections

This adds an option to `FlatPlatePaalmanPings` and `CylinderPaalmanPings` as follows:

`Emode = Efixed`, in which case the workspace unit requirement can be waived, and the correction will be computed for a single wavelength corresponding to the given Efixed. Fixed value itself will be by default read from the instrument parameters, but can be overridden by the user input.

For `FlatPlatePaalmanPings`, the `Emode` property existed before, but was not actually in use.
So the previously existing 3 options (`Elastic, Direct, Indirect`) will be equivalent and they are left for legacy, and for consistency with cylinder algorithm. Only the `Efixed` new option will result in different behaviour.

**To test:**

1. Create a workspace with instrument defined, but with x unit other than wavelength.
2. Run the above mentioned algorithms with the option`Emode=Efixed` and specify some value for `Efixed`. For the chemical formula, use for example `H2-O`.  The rest of the options can be left to defaults. The algorithms should run and produce the correction workspace which should have one bin corresponding to the wavelength = sqrt(81.787 / Efixed) Angstrom. 
3. Use `SetInstrumentParameter` to add a parameter called `Efixed` of type `Number` with some value to the workspace created at step 1.
4. Repeat step 2. without giving the Efixed value.



Fixes #18652 .

Release notes to be added to 3.10.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
